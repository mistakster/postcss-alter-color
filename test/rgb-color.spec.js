const process = require('./utils/process');

it('should alter rgb colors', () => {
  const source = `
div {
  color: rgb(0,0,0);
}
`;

  return process(source, {from: 'black', to: 'rgb(255,0,0)'})
    .andMatchSnapshot();
});

it('should replace an rgb color with the keyword', () => {
  const source = `
div {
  color: rgb(0,0,0);
}
`;

  return process(source, {from: 'black', to: 'red'})
    .andMatchSnapshot();
});

it('should replace an rgb color with the short hex color', () => {
  const source = `
div {
  color: rgb(0,0,0);
}
`;

  return process(source, {from: 'black', to: '#f00'})
    .andMatchSnapshot();
});

it('should replace an rgb color with the full hex color', () => {
  const source = `
div {
  color: rgb(0,0,0);
}
`;

  return process(source, {from: 'black', to: '#ff3fe0'})
    .andMatchSnapshot();
});

it('should replace an rgb color with the hsl color', () => {
  const source = `
div {
  color: rgb(0,0,0);
}
`;

  return process(source, {from: 'black', to: 'hsl(0,100%,50%)'})
    .andMatchSnapshot();
});
