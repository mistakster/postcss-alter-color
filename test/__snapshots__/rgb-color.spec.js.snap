// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should alter rgb colors 1`] = `
"
div {
  color: rgb(255,0,0);
}
"
`;

exports[`should replace an rgb color with the full hex color 1`] = `
"
div {
  color: #ff3fe0;
}
"
`;

exports[`should replace an rgb color with the hsl color 1`] = `
"
div {
  color: hsl(0,100%,50%);
}
"
`;

exports[`should replace an rgb color with the keyword 1`] = `
"
div {
  color: red;
}
"
`;

exports[`should replace an rgb color with the short hex color 1`] = `
"
div {
  color: #f00;
}
"
`;
